src/KeyboardAvoidingView.tsx

```diff
@@ -114,6 +141,12 @@ export default class KeyboardAvoidingView extends React.PureComponent<
     ) {
       marginBottom = keyboardHeight;
     }
-    return <View {...rest} style={[style, { marginBottom }]} />;
+
+    const defaultView = (
+      <View {...rest} style={[style, { marginBottom }]}>
+        {this.props.children}
+      </View>
+    );
+    return useSafeAreaView ? this.withSafeArea(defaultView) : defaultView;
```

Comment:
The implementation should be much simpler. Introducing a wrapping requires us to have `safeAreaStyle`, which is unnecessary.

```typescript
const { style, androidSoftInputMode, behavior, useSafeAreaView, ...rest } = this.props;
const { keyboardHeight } = this.state;
const isShowingKeyboard = keyboardHeight > 0;
let marginBottom = 0;

// ...

if (useSafeAreaView) {
  return <SafeAreaView
    {...rest}
    style={[style, { marginBottom }]}
    forceInset={{
      bottom: isShowingKeyboard ? "never" : "always",
    }}
  />;
}

return <View {...rest} style={[style, { marginBottom }]} />;

```